app/src/org/commcare/views/widgets/ComboboxWidget.java

```diff
@@ -0,0 +1,163 @@
+                        type, fontSize);
+        Combobox combobox = new Combobox(context, choices, adapter);
+        combobox.setTextSize(TypedValue.COMPLEX_UNIT_DIP, fontSize);
+        return combobox;
+    }
+
+    private static ComboboxAdapter getAdapterForComboboxWidget(Context context, String[] choices,
+                                                              ComboboxAdapter.FilterType type,
+                                                               int fontSize) {
+        choices = SpinnerWidget.getChoicesWithEmptyFirstSlot(choices);
+        ComboboxAdapter adapter = new ComboboxAdapter(context, choices, type);
+        adapter.setCustomTextSize(fontSize);
+        return adapter;
+    }
+
+    private void initChoices(FormEntryPrompt prompt) {
+        choices = prompt.getSelectChoices();
+        choiceTexts = new Vector<>();
+        for (int i = 0; i < choices.size(); i++) {
+            choiceTexts.add(prompt.getSelectChoiceText(choices.get(i)));
+        }
+    }
+
+    private void addListeners() {
+        comboBox.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                widgetEntryChanged();
+            }
+        });
+
+        // Note that Combobox has an OnFocusChangeListener defined in its own class, so when
+        // re-setting it here we have to make sure to do all of the same things that the original
+        // implementation does, in addition to the new behavior
+        comboBox.setOnFocusChangeListener(new OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (hasFocus) {
+                    comboBox.showDropDown();
+                } else {
+                    comboBox.autoCorrectCapitalization();
+                    widgetEntryChanged();
+                }
+            }
+        });
+
+        comboBox.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                clearWarningMessage();
+            }
+        });
+    }
+
+    private void fillInPreviousAnswer(FormEntryPrompt prompt) {
+        if (prompt.getAnswerValue() != null) {
+            String previousAnswerValue = ((Selection)prompt.getAnswerValue().getValue()).getValue();
+            for (int i = 0; i < choices.size(); i++) {
+                if (choices.get(i).getValue().equals(previousAnswerValue)) {
+                    comboBox.setText(choiceTexts.get(i));
+                    // setting this text will cause the dropdown to open up, which we don't want
```

Comment:
This came up in a couple of places and seems pretty awkward/unintuitive to me. I wonder if it might be worth putting the behavior more explicitly into the comboBox somehow. Is this a general awkwardness around the AutoCompleteTextView in general?